libyul/optimiser/BlockHasher.cpp

```diff
@@ -86,8 +107,7 @@ void BlockHasher::operator()(Identifier const& _identifier)
 void BlockHasher::operator()(FunctionCall const& _funCall)
 {
 	hash64(compileTimeLiteralHash("FunctionCall"));
```

Comment:
> You moved this to `hashFunctionCall()` so you should remove it from here (and from `ExpressionHasher`).

Ah yeah, missed that one. Thanks!

> why is the number of args hashed for user-defined functions but not for builtins

For builtins the id is (modulo dialect) enough to uniquely identify it. Also it doesn't require passing down the `Dialect` - which is what we would have to do, if I was also hashing the number of args.
For user-defined functions I am actually not sure why we would hash the number of arguments, I've just copied the existing implementation without thinking about it too much. 
Now I was thinking that it may be because of function overloading, but there is no function overloading in Yul. So perhaps we should remove it (especially since the arguments are visited anyways).

Reply:
Oh, good point about needing access to `Dialect`.

I wonder if it might have been a remainder of the old Ewasm dialect? Or some older situation were we did actually have some overloaded Yul builtins? Or a safeguard against comparing ASTs using different dialects?

In any case, I had basically the same reasoning so if we both agree, I think we can safely drop it.